Client/src/Pages/Settings Pages/StudentSettings/ProfileSettings/ConfirmImageDialog.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import {Dialog, DialogTitle, DialogContent, DialogActions, Button, Typography, CircularProgress} from '@material-ui/core'
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+
+export default class ConfirmImageModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            open: false,
+            picture: null,
+            loading: false
+        }
+    }
+
+    toggleOpen = () => {
+        this.setState({ open: !this.state.open });
+    }
+
+    async handleProfileChange(file) {
+        this.setState({
+            open: true, // Better to let the user know that image is being processed.
+            loading: true
+        });
+
+        file = await this.convertToJPEG(file);
+        
+        this.setState({
+            loading: false,
+            picture: file
+        });
+    }
+
+    handleImageUpload = () => {
+        this.fetchPresignedPostURL(this.state.picture);
+    }
+
+    fetchPresignedPostURL = (file) => {
+        fetch('https://7jdf878rej.execute-api.us-east-2.amazonaws.com/test/users/' + sessionStorage.getItem('id') + '/picture/upload')
+        .then((response) => response.json())
+        .then((presignedPostData) => {
+            const formData = new FormData();
+            Object.keys(presignedPostData.fields).forEach((field) => {
+                formData.append(field, presignedPostData.fields[field])
+            })
+            formData.append("file", file);
+            fetch(presignedPostData.url, {
+                method: 'POST',
+                body: formData
+            }) // We don't need to do any logic here for now.
+        })
+    }
+
+    convertToJPEG = (file) => {
```

Comment:
I'm surprised to see file conversion logic done on the client side. I would expect that there will be sufficient edge cases that a conversion like this is better handled on the backend. Ex - handling different file types, size/dimension limitations, compression, malformed files, etc.

But I'm not super familiar with what the requirements of this feature are, so maybe this is fine!

Reply:
So, we couldn't use a lambda function here but we had to use S3 pre-signed URLs here which directly uploads the file to the S3 storage. Now, I think we could use S3 triggers to convert the file on the backend once the file gets fully uploaded, but just for not making it complex right now, I included it here. We could potentially look into better ways to do this in the future.